### ISSUE TITLE
Ignoring "deprecated" errors in class file

### ISSUE THREAD

**Issue by mvar** [squizlabs/PHP_CodeSniffer#827]
Code:

```php
<?php

namespace Foo;

// @codingStandardsIgnoreStart
@trigger_error(
    'The XXX class is deprecated and will be removed in 2.0. Use YYY instead.',
    E_USER_DEPRECATED
);
// @codingStandardsIgnoreEnd

use Bar\YYY;

/**
 * @deprecated Will be removed in 2.0. Use the YYY instead.
 */
class XXX extends YYY
{
}
```

Code Sniffer generates warnings for `trigger_error ()` even when it is wrapped in `@codingStandardsIgnore`. Is there a way to silent these warnings?

**Comment by gsherwood** [comment]
I'm assuming you are talking about the side effects warning from PSR1/2. If that is the case, the error is reported on line 1 as it applies to the entire file and not a specific line. The only way to mute that error is to ignore the entire file or exclude it using the `--ignore` command line argument.

The reason for this is that wrapping code with the ignore start/end comments just mutes errors reported on those lines. It doesn't make those parts of the file invisible to checking.

To ignore the entire file, put the comment `// @codingStandardsIgnoreFile` somewhere near the top of your file (anywhere is actually file, but performance is better the higher you place it in the file).

Does that help?

**Comment by mvar** [comment]
Thanks for reply. I got it! I don't want to ignore whole file (because I have a lot of deprecated classes). I think it's better to skip Code Sniffer warnings (`-n`).

Lets talk a little bit more about deprecated classes. An idea to trigger error I took from [Symfony](https://github.com/symfony/symfony/blob/2.8/src/Symfony/Bundle/TwigBundle/TwigDefaultEscapingStrategy.php#L14) source code. I think that's really useful for end user. Sadly code like this does not comply PSR2. Mmm.. Maybe I need to address my issue to php-fig?

**Comment by gsherwood** [comment]
> I think it's better to skip Code Sniffer warnings (-n).

Yes, also a good option.

> Sadly code like this does not comply PSR2. Mmm.. Maybe I need to address my issue to php-fig?

That might be a good idea. I'm not aware of any existing PHP-FIG discussions around deprecation.

I'll close this issue now that you know what is going on. Thanks for getting back to me.

**Comment by mvar** [comment]
Thanks for answers.

If you are interested in this topic you can follow issue on php-fig/fig-standards#714
